From e01ead1385e34358efce0ca0da30e414010c79c9 Mon Sep 17 00:00:00 2001
From: Willy Zhang <willyzhang@google.com>
Date: Fri, 26 Dec 2025 20:37:38 -0800
Subject: [PATCH 1/3] Remove compile_data check for header-only support

Remove the check that fails if no static libraries are found.
This allows bindgen to work with header-only C++ libraries (or dynamic-only dependencies).

This patch is based on the one from the OpenTitan repository:
https://github.com/lowRISC/opentitan/blob/1f646a6566988315ee0215edfe8f10f0434ac498/third_party/rust/patches/rules_rust.bindgen_static_lib.patch

---
 private/bindgen.bzl | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/extensions/bindgen/private/bindgen.bzl b/extensions/bindgen/private/bindgen.bzl
--- private/bindgen.bzl
+++ private/bindgen.bzl
@@ -152,11 +152,6 @@ def _generate_cc_link_build_info(ctx, cc_lib):
                 linker_search_paths.append(lib.pic_static_library.dirname)
                 compile_data.append(lib.pic_static_library)
 
-    if not compile_data:
-        fail("No static libraries found in {}".format(
-            cc_lib.label,
-        ))
-
     rustc_flags_file = ctx.actions.declare_file("{}.rustc_flags".format(ctx.label.name))
     ctx.actions.write(
         output = rustc_flags_file,
